docs: warn against using `rebaseWhen=never` plus `prCreation=not-pending` (#13223)
renovatebot/renovate: docs/usage/configuration-options.md

It is also recommended to avoid `rebaseWhen=never` as it can result in conflicted branches with outdated PR descriptions and/or status checks.

> Avoid setting `rebaseWhen=never` and then also setting `prCreation=not-pending` as this results in Renovate not creating any new PRs (unless you request them from the Dependency Dashboard).

Reviewer: ```suggestion
Avoid setting `rebaseWhen=never` and then also setting `prCreation=not-pending` as this can prevent creation of PRs.
```
